crates/allocator/src/bump.rs

```diff
@@ -101,13 +120,36 @@ impl InnerAlloc {
                     return None;
                 }
 
-                prev_page.checked_mul(PAGE_SIZE)
+                // cannot overflow on this arch
+                Some(prev_page * PAGE_SIZE)
+            }
+            fn heap_start() -> usize {
+                extern "C" {
+                    static __heap_base: usize;
+                }
+                let heap_start =  unsafe { &__heap_base as *const usize as usize };
+                // if the symbol isn't found it will resolve to 0
+                // for that to happen the rust compiler or linker need to break or change
+                assert_ne!(heap_start, 0);
+                heap_start
+            }
+            fn heap_end() -> usize {
+                // cannot overflow on this arch
+                core::arch::wasm32::memory_size(0) * PAGE_SIZE
```

Comment:
Does `wasm32::memory_size()` change as we request more pages during execution?

Reply:
Yes. It it tells you the current size. But since we only call it for the first allocation it doesn't matter.